src/Grpc.AspNetCore.Server/Internal/CallHandlers/ClientStreamingServerCallHandler.cs

```diff
@@ -57,8 +70,22 @@ protected override async Task HandleCallAsyncCore(HttpContext httpContext)
 
                 service = activator.Create();
 
-                response = await _invoker(
+                ClientStreamingServerMethod<TRequest, TResponse> resolvedInvoker = delegate (IAsyncStreamReader<TRequest> resolvedRequestStream, ServerCallContext resolvedContext)
+                {
+                    return _invoker(
                     service,
+                    resolvedRequestStream,
+                    resolvedContext);
+                };
+
+                foreach (var interceptor in _interceptors)
+                {
+                    var interceptorInstance = (Interceptor)ActivatorUtilities.CreateInstance(httpContext.RequestServices, interceptor);
```

Comment:
We have IGrpcServiceActivator that allows users to change how services are created, i.e. override the default per-request creation to singletons. I feel like we have an equivalent for interceptors. Either introduce a new interface, or perhaps IGrpcServiceAcivator should be responsible for activating the service's interceptors as well.